sdk/cpp/core/src/gnmi_client.cpp

```diff
@@ -1,273 +1,325 @@
 }
 
-void gNMIClient::init_capabilities() 
+bool gNMIClient::has_gnmi_version(::gnmi::CapabilityResponse* response) 
 {
-	std::string cap;
-	while (std::getline(capfile, cap)) 
-	{
-		capabilities.push_back(cap);
-		//std::cout << cap;
-	}
+    if (!(response->gnmi_version().size() > 0)) 
+    {
+        YLOG_ERROR("Capabilities Received Without gNMI Version");
+        throw(YCPPError{"Capabilities Received Without gNMI Version"});
+        return false;
+    }
+    return true; 
 }
 
-bool gNMIClient::get_capabilities(const CapabilityRequest& request, CapabilityResponse* response) 
+void gNMIClient::parse_capabilities_modeldata(::gnmi::CapabilityResponse* response)
 {
-	grpc::ClientContext context;
-	grpc::Status status = stub_->Capabilities(&context, request, response);
-	if (!status.ok()) 
-	{
-	  std::cout << "Capabilities RPC Failed." << std::endl;
-	  return false;
-	}
-	if (!response->gnmi_version().size() > 0) 
-	{
-	  std::cout << "Server has not returned gnmi version" << std::endl;
-	  return false;
-	} else 
-	{
-		std::cout << "Receive capability response with gNMI version: " << response->gnmi_version() << std::endl;
-		for (unsigned int i = 0, n = response->supported_models_size(); i < n; i++) 
-		{
-			std::cout << "supported_models size" << response->supported_models_size() << std::endl;
-			std::cout << "supported_models";
-			response->supported_models(i);
-		}
-		for (int i = 0, n = response->supported_encodings_size(); i < n; i++) 
-		{
-			std::cout << "supported_models size" << response->supported_encodings_size() << std::endl;
-			std::cout << "supported_models";
-			response->supported_encodings(i);
-		}
-	}
-    return true;
+    ::gnmi::ModelData modeldata;
+    string cap, name, organization, version; 
+
+    for (int i = 0, n = response->supported_models_size(); i < n; i++) 
+    {
+        cap.clear();
+        modeldata = response->supported_models(i);
+        name = modeldata.name();
+        organization = modeldata.organization();
+        version = modeldata.version();
+        if(!(modeldata.name().empty()))
+        {
+            YLOG_INFO("Name: {}", name.c_str());
+            cap.append("?module=" + name);
+        }
+        if(!(modeldata.organization().empty())) 
+            YLOG_INFO("Organization: {}", organization.c_str());
+        if(!(modeldata.version().empty()))
+        {
+            YLOG_INFO("Version: {}", version.c_str());
+            cap.append("&revision=" + version);
+        }
+        capabilities.push_back(cap);
+        YLOG_INFO("              ------------       ");
+        YLOG_INFO("");
+    }   
+}
+
+void gNMIClient::parse_capabilities_encodings(::gnmi::CapabilityResponse* response)
+{
+    ::gnmi::Encoding encoding;
+    string encoding_value;
+
+    for (int i = 0, n = response->supported_encodings_size(); i < n; i++) 
+    {
+        encoding = response->supported_encodings(i);
+        switch(encoding)
+        {
+            case 0: encoding_value = "JSON"; break;
```

Comment:
Yes this is redundant, the values have been defined in the generated files. Updated in  [66c08bb](https://github.com/shgandhi/ydk-gen/commit/66c08bbe5f1873c0d1dac73d75f813a84c587cf8)